skyportal/handlers/api/group_admission_request.py

```diff
@@ -0,0 +1,217 @@
+            admission_request = GroupAdmissionRequest.query.get(admission_request_id)
+            if admission_request is None:
+                return self.error("Invalid admission request ID.")
+
+            response_data = admission_request.to_json()
+            response_data["user"] = admission_request.user
+
+            return self.success(data=response_data)
+
+        q = GroupAdmissionRequest.query
+        if group_id is not None:
+            q = q.filter(GroupAdmissionRequest.group_id == group_id)
+        admission_requests = q.all()
+        response_data = [
+            {**admission_request.to_dict(), "user": admission_request.user}
+            for admission_request in admission_requests
+        ]
+        return self.success(data=response_data)
+
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Create a new group admission request
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  groupID:
+                    type: integer
+                  userID:
+                    type: integer
+                required:
+                  - groupID
+                  - userID
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New group admission request ID
+        """
+        data = self.get_json()
+        user_id = data.get("userID")
+        group_id = data.get("groupID")
+        if user_id is None:
+            return self.error("Missing required parameter `userID`")
+        if group_id is None:
+            return self.error("Missing required parameter `groupID`")
+        try:
+            user_id = int(user_id)
+        except ValueError:
+            return self.error("Invalid `userID` parameter; unable to parse to int")
+        try:
+            group_id = int(group_id)
+        except ValueError:
+            return self.error("Invalid `groupID` parameter; unable to parse to int")
+        # Ensure that requesting user is either sysadmin or is the user in question
+        if (
+            self.associated_user_object.id != user_id
+            and not self.current_user.is_system_admin
+        ):
+            return self.error("Insufficient permissions")
+        if Group.query.get(group_id) is None:
+            return self.error("Invalid group ID")
+        if User.query.get(user_id) is None:
+            return self.error("Invalid user ID")
+        admission_request = GroupAdmissionRequest(
```

Comment:
We should exclude single-user groups, because now you can submit a request there.